PWR/PWR_RTC/Core/Src/main.c

```diff
@@ -0,0 +1,95 @@
+#include "stm32l053xx.h"
+
+//------ DEFINES
+//--- INPUTS
+#define USR_BT_PRESS		(!(GPIOA->IDR & GPIO_IDR_ID10_Msk))
+
+//--- OUTPUTS
+#define GREEN_LED_ON		(GPIOA->ODR |= GPIO_ODR_OD7_Msk)
+#define	GREEN_LED_OFF		(GPIOA->ODR &= ~GPIO_ODR_OD7_Msk)
+#define TOGGLE_GREEN_LED	(GPIOA->ODR ^=GPIO_IDR_ID7_Msk)
+
+//--- GENERAL
+#define RTC_30_SECONDS	32000	//used for RTC config, equals to 30 seconds (time spent in standy mode)
+#define RTC_KEY1		0xCA	//1st key to unlock RTC's registers (write to RTC_WPR)
+#define RTC_KEY2		0x53	//2nd key to unlock RTC's registers (write to RTC_WPR)
+
+int main(void)
+{
+	/*							   *
+	 * --- GPIO INITIALIZATION --- *
+	 *							   */
+	/* -> GPIOA
+	 * PA14 -> AF mode (SWD_IO)		| Low speed	| Push pull
+	 * PA13 -> AF mode (SWD_CK)		| Low speed	| Push pull
+	 * PA10	-> Input (User Button)	| Low speed	| Pull-up
+	 */
+	RCC->IOPENR	 |=	RCC_IOPENR_GPIOAEN;		//enable GPIOA clock
+
+	GPIOA->MODER &=	~GPIO_MODER_MODE10_Msk;	//set PA10 as input
+	GPIOA->PUPDR |= GPIO_PUPDR_PUPD10_0;	//enables PA10 pull-up
+
+	//Green led, debug purpose
+	GPIOA->MODER &= ~GPIO_MODER_MODE7_1;	//set PA7 as output
+	GPIOA->BSRR |= GPIO_BSRR_BR_7;			//resets the corresponding OD7 bit
+	GREEN_LED_ON;
+
+	/*									   *
+	 * --- STANDBY MODE INITIALIZATION --- *
+	 *									   */
+	RCC->APB1ENR |= RCC_APB1ENR_PWREN;	//enable PWR clock
+	SCB->SCR = SCB_SCR_SLEEPDEEP_Msk; 	//low-power mode = deep sleep
+	PWR->CR |= 	PWR_CR_PDDS;			//making sure we're entering standby mode
+	PWR->CR |= 	PWR_CR_CWUF		|		//clears WUF after 2 system clock cycles
+				PWR_CR_ULP		;		//ultra low power mode enable
+
+	//entering standy mode procedure
+	DBGMCU->CR |= DBGMCU_CR_DBG_STANDBY;	//this bit needs to be set if you're going to debug this code
+
+	/*							   *
+	 *  --- RTC INITIALIZATION --- *
+	 *							   */
+	// --- Resetting RTC registers (debug purpose)
+	RCC->CSR |= RCC_CSR_RTCRST;
+	while(!(RCC->CSR & RCC_CSR_RTCRST));
+	RCC->CSR &= ~RCC_CSR_RTCRST;
+	while((RCC->CSR & RCC_CSR_RTCRST));
+
+	PWR->CR  |=	PWR_CR_DBP;						//enable write access to the RTC and RCC CSR registers
+	RCC->CSR |=	RCC_CSR_RTCSEL_LSI	|			//sets LSI as RTC clock source (37 kHz)

```

Comment:
Just make sure this register don't need to be set in an order.

Reply:
It's ok enable all together!

Reply:
I took a quick look again on the RM (in RTC and RCC chapter) and couldn't find anything about this. But, since I'm doing `while(!(RCC->CSR & RCC_CSR_LSIRDY)); //poll until LSI is stable` after turning LSI on and enabling it as RTC clock source, I think it's ok to leave it as it is

Reply:
Yeah, it's ok!